modules/tinymce/src/plugins/code/test/ts/browser/CodeSanityTest.ts

```diff
@@ -1,66 +1,55 @@
-      FocusTools.sSetFocus('Focus dialog', docBody, dialogSelector),
-      Mouse.sClickOn(docBody, 'button.tox-button:contains(Save)'),
-      Waiter.sTryUntil('Dialog should close', UiFinder.sNotExists(docBody, dialogSelector))
-    ]));
-
-    Pipeline.async({}, [
-      Log.stepsAsStep('TBA', 'Code: Set content in empty editor and assert values', [
-        Mouse.sClickOn(SugarElement.fromDom(editor.getContainer()), toolbarButtonSelector),
-        UiFinder.sWaitForVisible('Waited for dialog to be visible', docBody, dialogSelector),
-        sAssertTextareaContent(''),
-        sSetTextareaContent('<em>a</em>'),
-        sSubmitDialog(docBody),
-        tinyApis.sAssertContent('<p><em>a</em></p>')
-      ]),
-
-      Log.stepsAsStep('TBA', 'Code: Reopen dialog and check textarea content is correct', [
-        Mouse.sClickOn(SugarElement.fromDom(editor.getContainer()), toolbarButtonSelector),
-        UiFinder.sWaitForVisible('Waited for dialog to be visible', docBody, dialogSelector),
-        sAssertTextareaContent('<p><em>a</em></p>')
-      ]),
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/code/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
 
-      Log.stepsAsStep('TBA', 'Code: Change source code and assert editor content changes', [
-        sSetTextareaContent('<strong>b</strong>'),
-        sSubmitDialog(docBody),
-        tinyApis.sAssertContent('<p><strong>b</strong></p>')
-      ])
-    ], onSuccess, onFailure);
-  }, {
+describe('browser.tinymce.plugins.code.CodeSanityTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
     plugins: 'code',
-    theme: 'silver',
     toolbar: 'code',
     base_url: '/project/tinymce/js/tinymce'
-  }, success, failure);
+  }, [ Plugin, Theme ]);
+
+  const dialogSelector = 'div[role="dialog"]';
+  const toolbarButtonSelector = '[role="toolbar"] button[aria-label="Source code"]';
+
+  const setTextareaContent = (content: string) => {
+    const textarea: HTMLTextAreaElement = document.querySelector('div[role="dialog"] textarea');
+    textarea.value = content;
+  };
+
+  const assertTextareaContent = (expected: string) => {
+    const textarea: HTMLTextAreaElement = document.querySelector('div[role="dialog"] textarea');
+    assert.equal(textarea.value, expected, 'Should have correct value');
+  };
+
+  it('TBA: Set content in empty editor and assert values', async () => {
+    const editor = hook.editor();
+    TinyUiActions.clickOnToolbar(editor, toolbarButtonSelector);
+    await TinyUiActions.pWaitForPopup(editor, dialogSelector);
+    assertTextareaContent('');
+    setTextareaContent('<em>a</em>');
+    TinyUiActions.submitDialog(editor, dialogSelector);
+    TinyAssertions.assertContent(editor, '<p><em>a</em></p>');
+  });
+
+  it('TBA: Reopen dialog and check textarea content is correct', async () => {
+    const editor = hook.editor();
+    TinyUiActions.clickOnToolbar(editor, toolbarButtonSelector);
```

Comment:
Is it safe to depend on the above test _always_ running before this one? In the past I've had bedrock (on refresh) just run the test you were in the middle of without running the ones before it — could that happen here and would it cause issues?

Reply:
I'm not sure I follow sorry... This used to depend on the previous steps, but it's now been changed to be independant.

Reply:
Ideally tests should be able to be reordered so this should probably be a part of the previous test case. It looks to me like it is still asserting based on content from the test before?

Reply:
Ahh, you're right I missed the `sAssertTextareaContent`. Will fix that now, thanks!

Reply:
Should be fixed now. I force pushed though sorry, since I have commits I want to keep separate.